refactor based on SonarQube recommendations

OGHO-SAYE opened the pull request:
PR: Fix SonarQube warnings in ShuttleInfoPopup component, refactor stylesheet and StartAndDestinationPoints component

This PR addresses several SonarQube warnings:

1. Added PropTypes validation for the `visible` prop and `shuttleDetails.error` property
2. Extracted nested ternary operation into a separate function for better readability
3. Improved code structure and formatting
4. Refactored the `footer container` stylesheet to replace nested ternary operators with an immediately invoked function with if else statements
5. Made refactoring adjustments for empty objects for startAndDestinationPoints Function and 
6. Extracted ternary operator into an independent statement
7. Removed unused import statements
8. Extracted some inline code implementation to external functionality `renderTabBarIcon`

These changes improve code quality while maintaining existing functionality. The conditional logic is now easier to read and maintain in both the component and stylesheet. Carmen, I made sure my refactor did not make any changes to shuttle functionalities, but please review because the shuttleInfoPopup component had a big refactor.

sbiothmane commented:
There are still 13 issues that are very easy fixes (unused imports and variables, commented code and duplicate names in styles). Can you add those fixes or I can do it if you want.

OGHO-SAYE commented:
Ok sounds good will get to it soon

OGHO-SAYE commented:
@sbiothmane In my latest commit, I have addressed most of the issues, I left some variables untouched because it looked like it could be needed for future implementation.
